lib/backpex/filters/boolean.ex

```diff
@@ -48,87 +50,115 @@ defmodule Backpex.Filters.Boolean do
     quote do
       use BackpexWeb, :filter
 
+      alias Backpex.Filters.Boolean, as: BooleanFilter
+
       @behaviour Backpex.Filters.Boolean
 
       @impl Backpex.Filter
-      def query(query, _attribute, []), do: query
-
       def query(query, attribute, value) do
-        Enum.reduce(value, nil, fn
-          v, nil ->
-            Map.get(predicates(), v)
-
-          v, p ->
-            dynamic(^p or ^Map.get(predicates(), v))
-        end)
-        |> maybe_query(query)
+        BooleanFilter.query(query, options(), attribute, value)
       end
 
-      defp maybe_query(nil, query), do: query
-      defp maybe_query(predicates, query), do: where(query, ^predicates)
-
       @impl Backpex.Filter
       def render(var!(assigns)) do
-        var!(assigns) =
-          var!(assigns)
-          |> assign(:label, option_value_to_label(options(), var!(assigns).value))
+        var!(assigns) = assign(var!(assigns), :options, options())
 
         ~H"""
-        <%= @label %>
+        <BooleanFilter.render options={@options} value={@value} />
```

Comment:
Can we do even less in the quote blocks? Ideally, we only call the function outside the block here ([see Credo](https://hexdocs.pm/credo/Credo.Check.Refactor.LongQuoteBlocks.html)). Then we would save one of the two Heex parts.

Reply:
I need to assign the options because they can only be accessed from within the quote block. However, I have simplified it as follows:
```elixir
@impl Backpex.Filter
def render(assigns) do
  assigns = assign(assigns, :options, options())
  BooleanFilter.render(assigns)
end
```